lib/swiss_village_directory/repository.rb

```diff
@@ -11,5 +11,23 @@ def initialize
         Village.new(row[0], row[1], row[2], row[3], row[4], row[5], row[6])
       end
     end
+
+    def find(find_params = {})
+      @villages.find_all { |v| filter_matched?(v, find_params) }
+    end
```

Comment:
You don't intend to `find_all` if you use `find`, don't you?

Also: what sense does the default parameter make?